bundles/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Platform.java

```diff
@@ -484,6 +485,7 @@ public final class Platform {
 
 	private static final String LINE_SEPARATOR_VALUE_LF = "\n"; //$NON-NLS-1$
 	private static final String LINE_SEPARATOR_VALUE_CRLF = "\r\n"; //$NON-NLS-1$
+	private static Charset defaultCharset;
```

Comment:
1. better "nativeCharset" - well i am not even sure about that name, since i read there may be 3 different native Encodings for a) filename in syscalls b) system.in/system.out and c) our use case
2. either final or volatile - final preferred: 
`private static final nativeCharset = getNativeCharset();`

Reply:
if you want have `getSystemCharset()`, why should the field be `nativeCharset`? 

But if we keep `getNativeEncoding()`, the filed can be `nativeEncoding`.

Note: I proposed the method name to be in sync with Java system property `native.encoding`.